src/common/libioencode/ioencode.c

```diff
@@ -64,6 +95,22 @@ json_t *ioencode (const char *stream,
     return rv;
 }
 
+static int decode_data_base64 (char *src,
+                               size_t srclen,
+                               char **datap,
+                               size_t *lenp)
+{
+    ssize_t rc;
+    size_t size = base64_decoded_length (srclen);
+    if (!(*datap = calloc (1, size)))
+        return -1;
+    if ((rc = base64_decode (*datap, size, src, srclen)) < 0)
+        return -1;
+    *lenp = rc;
+    return 0;
+}
+
+
 int iodecode (json_t *o,
```

Comment:
consideration b/c of this change, should `char **data` be `void **data` now?

Reply:
I chose not to change the function prototypes. Even `base64_encode()` takes `char *src` as the buffer to encode, so I'd be inclined to leave things as is unless we hit an issue (though maybe you see an issue I don't?)

Reply:
nope, was just for consideration.